tests/gemm/test_cute_dsl_blockscaled_gemm.py

```
@@ -259,6 +259,73 @@ def test_blockscaled_gemm_python_interface(
+    # Compliant output (N contiguous, expert dim outermost): no cross-expert leak.
+    def canonical_out():
+        return torch.zeros(E, m, n, dtype=torch.bfloat16, device=dev).permute(1, 2, 0)
+
+    base = run(canonical_out(), 0x3C)  # all scales = 1.0
+    perturbed = run(canonical_out(), 0x7F)  # expert 0 scale = NaN
+    assert not torch.isnan(perturbed[1]).any(), "NaN leaked into expert 1"
+    torch.testing.assert_close(base[1], perturbed[1], atol=0.0, rtol=0.0)
+
+    # Non-compliant output (expert dim innermost) must be rejected, not silently
+    # corrupt results across experts.
+    with pytest.raises(ValueError, match="cross-expert|c_major|contiguous"):
```

Ruff (0.15.15)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)